docs/userguide.md

```diff
@@ -0,0 +1,192 @@
++ TiSpark runs in any Spark mode such as YARN, Mesos, and Standalone.
+
+
+## Recommended configuration
+
+### Deployment of TiKV and TiSpark clusters
+
+#### Configuration of the TiKV cluster
+
+For independent deployment of TiKV and TiSpark, it is recommended to refer to the following recommendations
+ 
++ Hardware configuration
+ - For general purposes, please refer to the TiDB and TiKV hardware configuration [recommendations](https://github.com/pingcap/docs/blob/master/op-guide/recommendation.md#deployment-recommendations).
+ - If the usage is more focused on the analysis scenarios, you can increase the memory of the TiKV nodes to at least 64GIf using  Hard Disk Drive (HDD), it is recommended to use at least 8 disks.
+
++ TiKV parameters (default)
+
+```
+[Server]
+End-point-concurrency = 8 # For OLAP scenarios, consider increasing this parameter
+[Raftstore]
+Sync-log = false
+
+
+[Rocksdb]
+Max-background-compactions = 6
+Max-background-flushes = 2
+
+[Rocksdb.defaultcf]
+Block-cache-size = "10GB"
+
+[Rocksdb.writecf]
+Block-cache-size = "4GB"
+
+[Rocksdb.raftcf]
+Block-cache-size = "1GB"
+
+[Rocksdb.lockcf]
+Block-cache-size = "1GB"
+
+[Storage]
+Scheduler-worker-pool-size = 4
+```
+
+#### Configuration of the independent deployment of the Spark cluster and the TiSpark cluster 
+
+ 
+Please refer to the [Spark official website](https://spark.apache.org/docs/latest/hardware-provisioning.html) for the detail hardware recommendations.
+
+The following is a short overview of the TiSpark configuration.
+
+Generally, it is  recommended to allocat 32G memory for Spark. Please reserve at least 25% of the memory for the operating system and buffer cache.
+
+It is recommended to provision at least 8 to 16 cores on per machine for Spark. Initially, you can assign all the CPU cores to Spark.
+
+Please refer to the Spark official configuration website at (https://spark.apache.org/docs/latest/spark-standalone.html). The following is an example based on the spark-env.sh configuration:
+
+```
+SPARK_EXECUTOR_MEMORY = 32g
+SPARK_WORKER_MEMORY = 32g
+SPARK_WORKER_CORES = 8
+```
+
+#### Hybrid deployment configuration for the TiSpark and TiKV cluster
+
+For the  hybrid deployment of TiSpark and TiKV, add the TiSpark required resources to the TiKV reserved resources, and allocate 25% of the memory for the system.
+ 
+
+## Deploy TiSpark
+
+Download the TiSpark's jar package [here] (https://download.pingcap.org/tispark-0.1.0-beta-SNAPSHOT-jar-with-dependencies.jar).
+
+### Deploy TiSpark on the existing Spark cluster
+
+Running TiSpark on an existing Spark cluster does not require a reboot of the cluster. You can use Spark's `--jars` parameter to introduce TiSpark as a dependency:
+
+```
+Spark-shell --jars $ PATH / tispark-0.1.0.jar
+```
+
+If you want to deploy TiSpark as a default component, simply place the TiSpark jar package into the jars path for each node of the Spark clusters and restart the Spark clusters:
```

Comment:
My understanding is that there are many nodes (machines).  TiSpark deploys package to each node of the Spark cluster and restarts node one by one. is that right?